### PR TITLE
Upgrade to atom-linter v2

### PR DESCRIPTION
**Pull request by ghost**
Fixed bug with relative path
Changed config and issue objects for v2
Removed coala.less (now using atom styles instead of our ones) 

Closes https://github.com/coala/coala-atom/issues/59

### CODE REVIEW

**Comment by Udayan12167** [comment]
I think your first commit should be split into three commits as it changes three different files?

**Comment by ghost** [comment]
May be, i'll do that way next time

File: lib/main.coffee [hunk 1]
```diff
@@ -43,12 +43,12 @@ module.exports =
...
       grammarScopes: ['*']
       scope: 'file'
-      lintOnFly: true
+      lintsOnChange: true
       lint: (textEditor) =>
         filePath = textEditor.getPath()
         parameters = []
         parameters.push '--find-config'
-        parameters.push '--limit-files=' + filePath
+        parameters.push '--limit-files=' + path.basename(filePath)
```
Comment (sils) [inline lib/main.coffee:51]: why the basename? An absolute path will always work, the basename may well not be working?

File: lib/main.coffee [hunk 2]
```diff
@@ -43,12 +43,12 @@ module.exports =
       name: 'coala'
       grammarScopes: ['*']
       scope: 'file'
-      lintOnFly: true
+      lintsOnChange: true
```
Comment (sils) [inline lib/main.coffee]: actually why was this on true? We only want on save right?

File: lib/main.coffee [hunk 3]
```diff
@@ -22,9 +22,9 @@ module.exports =
              '-(?<type>\\d+)-(?<message>.*)'
 
     @resultSeverity = [
-      'Info',    # Result severity 0 = Info
-      'Normal',  # Result severity 1 = Normal
-      'Major'    # Result severity 2 = Major
+      'info',    # Info
+      'warning',  # Normal
```
Comment (sils) [inline lib/main.coffee]: comment not aligned with the others

**Review by sils — CHANGES_REQUESTED** [review]
FWIW I'd just squash them together,it's a good logical change.

File: lib/main.coffee (hunk 1 above)
Comment (ghost) [reply lib/main.coffee:51]: but we execute with project path as workingDirectory for process, so we dont need absolute path

File: lib/main.coffee (hunk 2 above)
Comment (ghost) [reply lib/main.coffee]: Pls, look at this https://github.com/coala/coala-atom/issues/62

File: lib/main.coffee (hunk 1 above)
Comment (sils) [reply lib/main.coffee:51]: the basename is usually the part after the last slash right? So it's not necessarily relative to the project directory.

File: lib/main.coffee (hunk 2 above)
Comment (sils) [reply lib/main.coffee]: ah because of the file opening, makes sense :+1:

File: lib/main.coffee (hunk 1 above)
Comment (ghost) [reply lib/main.coffee:51]: as i said, coala will be executed with workingDirectory = directory, which contains file for analysis. So, we need to pass only file name in this direcory

File: lib/main.coffee [hunk 4]
```diff
@@ -22,9 +22,9 @@ module.exports =
              '-(?<type>\\d+)-(?<message>.*)'
 
     @resultSeverity = [
-      'info',    # Info
+      'info',     # Info
       'warning',  # Normal
-      'error'    # Major
+      'error'     # Major
```
Comment (satwikkansal) [inline lib/main.coffee]: this indentation change not relevant to the commit.

**Comment by satwikkansal** [comment]
Please describe what exactly are you fixing in the commit message of https://github.com/coala/coala-atom/pull/63/commits/4e4c3d229d674dfb9fecb2aecc67b57af0762946

**Comment by sils** [comment]
I just squashed it with a new commit message because it's not too important here and it's way more important to get this merged

**Pull request merged**